spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+
+
+describe SessionsController do
+  describe "GET new" do 
+    it "renders the new template for unauthenticated users" do
+      get :new 
+      expect(response).to render_template :new
+    end
+
+    it "rendirects to the home page for authinticated users" do
+      session[:user_id] = Fabricate(:user).id
+      get :new
+      expect(response).to redirect_to home_path
+    end
+  end
+
+  describe "POST create " do
+    
+    context "with valid credentials" do
+      it "puts the signed user in the session" do 
+        lana = Fabricate(:user)
+        post :create, email: lana.email, password: lana.password
+        expect(session[:user_id]).to eq(lana.id)
+      end
+      it "redirects to the home page " do
+        lana = Fabricate(:user)
+        post :create, email: lana.email, password: lana.password
+        expect(response).to redirect_to home_path
+      end
+      it "sets the notice" do 
+        lana = Fabricate(:user)
+        post :create, email: lana.email, password: lana.password
+        expect(flash[:success]).not_to be_blank
```

Comment:
You always want to use the successful version if possible. In this case that means `.to be_present`.